bee-protocol/src/workers/peer/manager_res.rs

```diff
@@ -48,64 +55,95 @@ type PeerTuple = (Arc<Peer>, Option<(GossipSender, oneshot::Sender<()>)>);
 pub struct PeerManagerInner {
     peers: HashMap<PeerId, PeerTuple>,
     // This is needed to ensure message distribution fairness as iterating over a HashMap is random.
-    // TODO private
-    pub(crate) keys: Vec<PeerId>,
+    keys: Vec<PeerId>,
 }
 
 #[derive(Default)]
-// TODO private
-pub struct PeerManager(pub(crate) RwLock<PeerManagerInner>);
```

Comment:
This feels like it was always a bad idea :eyes:

Reply:
I agree.

Reply:
Yeah I think we should have some guidelines or compilation checks about locks because I believe that every time we had an issue with the node freezing or not shutting down correctly is because we are using some `RwLock` or `Mutex` in an unstructured way.